taskcluster/async/asyncclient.py

```diff
@@ -0,0 +1,100 @@
+"""This module is used to interact with taskcluster rest apis"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import asyncio
+import aiohttp
+
+from contextlib import contextmanager
+
+import taskcluster.exceptions as exceptions
+import taskcluster.baseclient as baseclient
+import taskcluster.utils as utils
+import taskcluster.asyncutils as asyncutils
+
+log = logging.getLogger(__name__)
+
+
+class AsyncClient(baseclient.BaseClient):
+    """Async class for API Client Classes. Each individual Client class
+    needs to set up its own methods for REST endpoints and Topic Exchange
+    routing key patterns.
+    """
+
+    @contextmanager
+    def contextSession(self, *args, **kwargs):
+        yield self.createSession(*args, **kwargs)
+
+    def createSession(self, session=None, args=(), kwargs=None):
```

Comment:
Nit, but wouldn't it be preferable to prefix these methods with an underscore, just so it's clear that these are intended to be used internally. Ideally, it would also make docs better...
(I'm not python expert, so I defer to you)